### PR TITLE
M-TAN Authentication

### PR DESCRIPTION
**Pull request by martininka**
added authentication via M-TAN (SMS)

### CODE REVIEW

**Comment by senshi-x** [comment]
Thanks for the PR! looking good. Especially if handling mTAN really is that easy. Can't test it myself.

Not 100% sure about the shebang, though: Sadly, that is kinda OS-specific. I know of some Linux OS that use /usr/local/bin/python for their python3 setup. But I guess most at least have a symlink for python3 as well. Anyone have more insight on this?

**Comment by retiredHero** [comment]
the documentation says: 
tldr; better: "#!/usr/bin/env python3", but in some Unix setups hardcoding is needed (like /usr/bin/python3).
So there is no "right" answer to this.


long:

"The /usr/bin/env form of shebang line has one further special property. Before looking for installed Python interpreters, this form will search the executable PATH for a Python executable. This corresponds to the behaviour of the Unix env program, which performs a PATH search."

Source: 
https://docs.python.org/3.7/using/windows.html#shebang-lines

and:
put an appropriate Shebang line at the top of the script. A good choice is usually 
#!/usr/bin/env python3
which searches for the Python interpreter in the whole PATH. However, some Unices may not have the env command, so you may need to hardcode /usr/bin/python3 as the interpreter path.

Source:
https://docs.python.org/3.7/using/unix.html#miscellaneous

**Comment by martininka** [comment]
Ok it was a bad idea to combine two changes in one PR :)

Anyone able to test M-TAN? For sure it works for me :)

**Comment by martininka** [comment]
@senshi-x: Do you want me to remove shebang, so that we only have on "feature request" to discuss?

File: main.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#!/usr/local/bin/python3
```
Comment (senshi-x) [inline main.py:1]: change to #!/usr/bin/env python3, as per best practice recommendations

**Review by senshi-x — CHANGES_REQUESTED** [review]
Just change the shebang to the recommended, and it's good to merge :) .

**Review by senshi-x — APPROVED** [review]

**Comment by senshi-x** [comment]
Thanks again for the nice PR! <3

**Comment by martininka** [comment]
de Rien :)

Thanks for your excellent work! Love this project!
